src/pycram/designators/specialized_designators/action/dual_arm_pickup_action.py

```diff
@@ -0,0 +1,88 @@
+        """
+        super().__init__(object_designator_description,
+                         arms=[Arms.LEFT, Arms.RIGHT],
+                         grasps=grasps,
+                         resolver=resolver,
+                         ontology_concept_holders=ontology_concept_holders)
+
+        self.object_designator_description: Union[
+            ObjectDesignatorDescription, ObjectDesignatorDescription.Object] = object_designator_description
+
+        left_gripper_frame = World.robot.get_link_tf_frame(
+            RobotDescription.current_robot_description.kinematic_chains.get('left').get_tool_frame())
+        right_gripper_frame = World.robot.get_link_tf_frame(
+            RobotDescription.current_robot_description.kinematic_chains.get('right').get_tool_frame())
+        self.gripper_list: List[str] = [left_gripper_frame, right_gripper_frame]
+
+
+    def ground(self) -> PickUpActionPerformable:
+        if isinstance(self.object_designator_description, ObjectDesignatorDescription.Object):
+            obj_desig = self.object_designator_description
+        else:
+            obj_desig = self.object_designator_description.resolve()
+
+        rospy.loginfo("Calculating closest gripper to object {}".format(obj_desig.name))
+
+        local_transformer = LocalTransformer()
+
+        object_frame = obj_desig.world_object.tf_frame
+        distances = []
+        # Iterate over possible grippers
+        for gripper in self.gripper_list:
+            # Object pose in gripper frame
+            object_T_gripper: Transform = local_transformer.lookup_transform_from_source_to_target_frame(gripper,
+                                                                                            object_frame)
+            object_V_gripper: Vector3 = object_T_gripper.translation  # translation vector
+            distance = norm(array([object_V_gripper.x, object_V_gripper.y, object_V_gripper.z]))
+            rospy.loginfo(f"Distance between {gripper} and {obj_desig.name}: {distance}")
+            distances.append(distance)
+
+        min_index = distances.index(min(distances))
+        winner_frame = self.gripper_list[min_index]
+
+        if winner_frame == World.robot.get_link_tf_frame(
+                RobotDescription.current_robot_description.kinematic_chains.get('left').get_tool_frame()):
+            winner = Arms.LEFT
+            rospy.loginfo("Returning left arm as the winner")
+        elif winner_frame == World.robot.get_link_tf_frame(
+                RobotDescription.current_robot_description.kinematic_chains.get('right').get_tool_frame()):
+            winner = Arms.RIGHT
+            rospy.loginfo("Returning right arm as the winner")
+        else:
+            raise ValueError(f"Could not determine the winner arm for the pickup action. Winner: {winner_frame}")
+
+        return PickUpActionPerformable(object_designator=obj_desig, arm=winner, grasp=self.grasps[0])
```

Comment:
Just a general thing, you implement this in the ground method which is the default for reasoning. It would be better to move your code to the \_\_iter\_\_  method and yield the results according to the calculated distances.